Mathlib/Logic/Embedding/Basic.lean

```diff
@@ -0,0 +1,482 @@
+/-
+Copyright (c) 2017 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Mario Carneiro
+-/
+import Mathlib.Data.Prod.PProd
+import Mathlib.Logic.Equiv.Basic
+
+/-!
+# Injective functions
+-/
+
+
+universe u v w x
+
+namespace Function
+
+-- port note: in Lean 3 this was tagged @[nolint has_nonempty_instance]
+/-- `α ↪ β` is a bundled injective function. -/
+structure Embedding (α : Sort _) (β : Sort _) where
+  /-- An embedding as a function. Use coercion instead. -/
+  toFun : α → β
+  /-- An embedding is an injective function. Use `Function.Embedding.injective` instead. -/
+  inj' : Injective toFun
+#align function.embedding Function.Embedding
+
+/-- An embedding, a.k.a. a bundled injective function. -/
+infixr:25 " ↪ " => Embedding
+
+instance {α : Sort u} {β : Sort v} : EmbeddingLike (α ↪ β) α β where
+  coe := Embedding.toFun
+  injective' := Embedding.inj'
+  coe_injective' f g h := by { cases f; cases g; congr }
+
+/-- See Note [custom simps projection] -/
+def Embedding.Simps.apply {α β} (e : α ↪ β) : α → β := e
+
+initialize_simps_projections Embedding (toFun → apply)
+
+-- porting note: this needs `tactic.lift`.
+--instance {α β : Sort _} : CanLift (α → β) (α ↪ β) coeFn Injective where prf f hf := ⟨⟨f, hf⟩, rfl⟩
+
+end Function
+
+section Equiv
+
+variable {α : Sort u} {β : Sort v} (f : α ≃ β)
+
+/-- Convert an `α ≃ β` to `α ↪ β`.
+
+This is also available as a coercion `equiv.coe_embedding`.
+The explicit `equiv.to_embedding` version is preferred though, since the coercion can have issues
```

Comment:
```suggestion
This is also available as a coercion `equiv.coeEmbedding`.
The explicit `equiv.toEmbedding` version is preferred though, since the coercion can have issues
```